gatsby/content/blog/2022/07/2022-08-03-synapse-1.64.mdx

```diff
@@ -0,0 +1,98 @@
+---
+date: '2022-08-03'
+title: Synapse 1.64 released
+categories:
+- Releases
+author: Brendan Abolivier
+---
+
+It's that time again: there's a new Synapse release, fresh out of the oven!
+Let's take a look at what's inside [Synapse
+1.64](https://github.com/matrix-org/synapse/releases/tag/v1.64.0).
+
+## Delegating email verification is now deprecated
+
+[Synapse 1.4.0](https://github.com/matrix-org/synapse/releases/tag/v1.4.0)
+introduced a configuration option (`account_threepid_delegates.email`) to allow
+homeservers to delegate validating the ownership of email addresses to an
+external identity server. This validation is used by Synapse when associating
+email addresses to a Matrix account, or before performing a password reset.
+
+As of Synapse 1.64, this option is deprecated. This is because this option
+relies on old API endpoints that have since been removed from the Matrix
+specification. Server administrators who still need the ability to validate the
+ownership of email addresses must configure an SMTP server in their Synapse
+configuration file. See the [configuration
+guide](https://matrix-org.github.io/synapse/v1.64/usage/configuration/config_documentation.html#email)
+for more information.
+
+We plan to fully remove this configuration option in Synapse 1.66, which is
+expected to be released in 4 weeks.
+
+Note that the equivalent option to validate the ownership of phone numbers
+(`account_threepid_delegates.msisdn`) can still be used, though we expect to
+deprecate it in a future release of Synapse.
+
+## Improved TLS support for sending emails
+
+When configuring an SMTP server to use to send out emails to users, server
+administrators can configure Synapse to use TLS to communicate to that server.
+Until now, only [STARTTLS](https://en.wikipedia.org/wiki/Opportunistic_TLS) was
+supported in this case, and a long-standing pain point for server administrators
+has been the lack of support for implicit TLS (i.e. the use of TLS without a
+STARTTLS upgrade).
```

Comment:
Is there a relevant hyperlink here to a wikipedia page or RFC we can use?

Should `implicit TLS` be enclosed in scare quotes (`"implicit TLS"`)? We didn't think the naming was very good and used a different name in the config options.

Reply:
maybe put it in `*implicit TLS*` italics.

Implicit TLS is a well-known term amongst SMTP server admins I think, but probably less so everyone else. Trouble is that there isn't a great word for it otherwise either; many people call it 'SSL' as opposed to 'TLS', but that's so wrong I wouldn't want to encourage it :-P.

https://en.wikipedia.org/wiki/Simple_Mail_Transfer_Protocol#STARTTLS_or_%22Opportunistic_TLS%22 has a bit of context

Reply:
> Implicit TLS

I think we discussed this in the changelog a bit already, but I very much do not think it is a well known term. 😢

Reply:
I have seen the terminology 'Implicit' and 'Explicit' TLS in FTP clients as well.

Maybe to help clarify it you can mention the standard port number which is 465.

Regardless of whether 'Implicit TLS' is a well-known term or not, it's a standard phrase that you can search for and typically the people who actually care about it will know what it means.

Reply:
agreed. I'm also deeply unconvinced it was a long-standing pain point; STARTTLS is fine for everyone except the one guy who wrote the patch.

Let's drop the second half of this sentence ("and a long-standing...")  and replace the next paragraph with something like 'Synapse 1.64 adds the `force_tls` option, which means it will use TLS for the initial connection rather than upgrading via STARTTLS'.

Reply:
> I'm also deeply unconvinced it was a long-standing pain point

To clarify this: my reasoning was that the issue has been opened for 2 years and several people have had the same complaint (on the issue, and I think I've seen it mentioned by others in other places?). But happy to drop it.